tutorials/Oracles/Oracles.ipynb

```diff
@@ -0,0 +1,1015 @@
+    "    That's right!  If we apply a Hadamard operator to the third qubit, we will be able to distinguish between the input state and the output state. $$(I\\otimes I \\otimes H)|11+\\rangle = |110\\rangle$$ $$(I\\otimes I \\otimes H)|11-\\rangle = |111\\rangle$$ Now if we were to measure the input state versus the output state we would detect that the phase from $|y\\rangle=|-\\rangle$ was kicked back into our input state $|x\\rangle$!\n",
+    "</details>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we will explore how you could create a oracle converter\n",
+    "\n",
+    "#### <span style=\"color:blue\">Exercise 4</span>: Apply the phase oracle\n",
+    "* **Input:** \n",
+    "    * a marking oracle\n",
+    "    * a list of qubits to apply the phase oracle with\n",
+    "* After execution, the list of qubits will have had a phase oracle applied to them, where this phase oracle implements the same function $f(x)$ as the marking oracle does.\n",
+    "\n",
+    "> An oracle converter is useful because many quantum algorithms rely on a phase oracle, such as Grover's algorithm, however it is often easier to implement a marking oracle.  This converter will provide us a way to convert a marking oracle we are interested in into a phase oracle; which could then be leveraged in a quantum algorithm such as Grover's algorithm.\n",
+    "\n",
+    "<br/>\n",
+    "<details>\n",
+    "  <summary><b>If you're stuck on where to begin, click here!</b></summary>\n",
+    "    Recall that you can allocate extra qubits to assist in any operation.  Is there a state that you could prepare with an auxiliary qubit which would enable you to flip the phase of the input state $|x\\rangle$ subject to the function $f(x)$ for the marking oracle?\n",
+    "</details>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%check_kata E4_Apply_Phase_Oracle\n",
+    "\n",
+    "operation Apply_Phase_Oracle(markingOracle: ((Qubit[], Qubit) => Unit is Adj + Ctl), qubits: Qubit[]) : Unit\n",
+    "is Adj + Ctl {\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### <span style=\"color:blue\">Demo 3</span>: Oracle Converter\n",
+    "* **Input:**\n",
+    "    * a marking oracle\n",
+    "* **Output:**\n",
+    "    * a phase oracle which takes a register as input and flips the phase of that register according to the function $f(x)$ that the marking oracle implements\n",
+    "\n",
+    "In this demo we will use the oracle converter to convert your previously implemented marking oracle, $U_{7,mark}$ into a phase oracle.\n",
+    "\n",
+    "> **You must have Exercises 2, 3 and 4 completed correctly for this demo to behave correctly!**"
+   ]
+  },
+  {
```

Comment:
This is the demo that is causing CI build to fail; the magic commands is implemented in such a way that the operations are not defined when ran using `%check_kata` but are defined when ran using `%kata`, so the demo will work if the learner does the exercises but will fail if CI build tries to validate it... I think for now we can exclude this cell from validation, as described [here](https://github.com/microsoft/QuantumKatas/blob/main/.github/CONTRIBUTING.md#excluding-individual-tasks-from-validation) (probably for `multicell_solution` reason)

Reply:
This is a super cool feature